app/controllers/katello/api/v2/host_tracer_controller.rb

```diff
@@ -2,50 +2,28 @@ module Katello
   class Api::V2::HostTracerController < Api::V2::ApiController
     before_action :find_host, :only => :index
 
+    resource_description do
+      api_version 'v2'
+      api_base_url "/api"
+    end
+
     api :GET, "/hosts/:host_id/traces", N_("List services that need restarting on the host")
     param :host_id, :number, :required => true, :desc => N_("ID of the host")
     def index
       collection = scoped_search(index_relation, :application, :asc, :resource_class => ::Katello::HostTracer)
       respond_for_index(:collection => collection)
     end
 
-    api :PUT, "/traces/resolve", N_("Resolve Traces")
+    api :PUT, "/hosts/:host_id/traces/resolve", N_("Resolve Traces")
+    param :host_id, :number, :required => true, :desc => N_("ID of the host")
     param :trace_ids, Array, :required => true, :desc => N_("Array of Trace IDs")
     def resolve
       traces = Katello::HostTracer.resolvable.where(id: params[:trace_ids])
+      result = Katello::HostTraceManager.resolve_traces(traces)
+
+      task = ForemanTasks::Task.find(result.first.job_invocation.task_id)
 
-      traces.each do |trace|
-        if trace.reboot_required?
-          trace.helper = 'reboot'
-        end
-      end
-
-      traces_by_host_id = traces.group_by(&:host_id)
-      traces_by_helper = traces.group_by(&:helper)
-
-      composers = []
-
-      if traces_by_host_id.size < traces_by_helper.size
-        traces_by_host_id.each do |host_id, trace|
-          needed_traces = trace.map(&:helper).join(',')
-          joined_helpers = { :helper => needed_traces }
-          composers << JobInvocationComposer.for_feature(:katello_service_restart, [host_id], joined_helpers)
-        end
-      else
-        traces_by_helper.each do |helper, trace|
-          helpers = { :helper => helper }
-          composers << JobInvocationComposer.for_feature(:katello_service_restart, trace.map(&:host_id), helpers)
-        end
-      end
-
-      job_invocations = []
-
-      composers.each do |composer|
-        composer.trigger
-        job_invocations << composer.job_invocation
-      end
-
-      render json: job_invocations
+      render json: task
```

Comment:
It looks like we have to define a task instead if we use this:
```ruby
task = async_task(::Actions::Katello::ContentView::Promote,
                        @version, @environments, is_force, params[:description])
respond_for_async :resource => task
```

Since we are having the service class and the Composer take care of that, I think we should keep it as is. Also the way we are returning the task ID, we eliminate the need for the hammer-foreman-tasks pr that I had originally opened. 

Given the time crunch and with it working currently with hammer/ui, I vote for keeping it the way it is now, and when we address new issues, we can experiment with it there. I am happy to make a redmine for it so we have record of it.